cache/blobs.go

```diff
@@ -106,13 +107,23 @@ func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool
 			if release != nil {
 				defer release()
 			}
-			desc, err := sr.cm.Differ.Compare(ctx, lower, upper,
-				diff.WithMediaType(mediaType),
-				diff.WithReference(sr.ID()),
-				diff.WithCompressor(compressorFunc),
-			)
-			if err != nil {
-				return nil, err
+			var desc ocispecs.Descriptor
+			if !isTypeWindows(sr) {
+				// Try optimized diff for overlayfs
+				desc, err = sr.computeOverlayBlob(ctx, lower, upper, mediaType, sr.ID(), compressorFunc)
+				if err != nil {
+					logrus.Debugf("failed to compute blob (%s) from diff of overlay snapshotter: %+v", sr.ID(), err)
```

Comment:
we should avoid printing this if overlay is not used. Maybe `tryComputeOverlayBlob` and return `desc, ok, err` so we can determine between actual error and mount just not being overlay.

Reply:
Thank you for the review. Fixed the error handling logic.